clients/tabby-agent/src/http/tabbyApiClient.ts

```diff
@@ -353,6 +367,7 @@ export class TabbyApiClient extends EventEmitter {
       canceled: false,
       timeout: false,
       notAvailable: false,
+      rateLimited: false,
```

Comment:
Don't add rateLimited in statsData, call `updateIsRateLimited` directly in the error handling branches.